LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/OTAAHelper/OTAAKeysGenerator.cs

```diff
@@ -10,19 +10,15 @@ namespace LoRaTools
 
     public static class OTAAKeysGenerator
     {
-        private static readonly RandomNumberGenerator RndKeysGenerator = RandomNumberGenerator.Create();
-
-        public static string GetNwkId(byte[] netId)
+        public static DevAddr GetNwkId(byte[] netId)
         {
             if (netId is null) throw new ArgumentNullException(nameof(netId));
 
-            var nwkPart = netId[0] << 1;
-            var devAddr = new byte[4];
-
-            RndKeysGenerator.GetBytes(devAddr);
-
-            devAddr[0] = (byte)((nwkPart & 0b11111110) | (devAddr[0] & 0b00000001));
-            return ConversionHelper.ByteArrayToString(devAddr);
+            Span<byte> bytes = stackalloc byte[4];
+            RandomNumberGenerator.Fill(bytes);
+            var id = netId[0] & 0b0111_1111;
+            var address = BitConverter.ToInt32(bytes) & 0x01ff_ffff;
```

Comment:
Instead of relying on "magical" bit masks here, should we consider making the `NetworkAddressMask` and `NetworkIdMask` `public` or at least internal? It would be easier to figure out what's going on in such scenarios then

Reply:
This is not the same mask. Here, the 7 LSB of the NetID become the 7 MSB of the `DevAddr`. Perhaps the mask can be introduced when we get round to using `NetId` as part of #972.